RadarSDK/Radar.m

```diff
@@ -340,84 +348,65 @@ + (void)rejectEventId:(NSString *)eventId {
 
-+ (void)logConversionWithName:(NSString *)name
-                     metadata:(NSDictionary *_Nullable)metadata
-            completionHandler:(RadarLogConversionCompletionHandler)completionHandler {
-    [self trackOnceWithCompletionHandler:^(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarEvent *> * _Nullable events, RadarUser * _Nullable user) {
++ (void)sendLogConversionRequestWithType:(NSString * _Nonnull) type
+                                metadata:(NSDictionary * _Nullable) metadata
+                       completionHandler:(RadarLogConversionCompletionHandler) completionHandler {
+    [[RadarAPIClient sharedInstance] sendEvent:type withMetadata:metadata completionHandler:^(RadarStatus status, NSDictionary * _Nullable res, RadarEvent * _Nullable event) {
         if (status != RadarStatusSuccess) {
             if (completionHandler) {
                 [RadarUtils runOnMainThread:^{
-                    completionHandler(status, nil, nil, nil);
+                    completionHandler(status, nil);
                 }];
             }
-
+            
             return;
         }
+        
+        if (completionHandler) {
+            [RadarUtils runOnMainThread:^{
+                completionHandler(status, event);
+            }];
+        }
+    }];
+}
 
-        [[RadarAPIClient sharedInstance] sendEvent:name withMetadata:metadata user:user trackEvents:events completionHandler:^(RadarStatus status, NSDictionary * _Nullable res, NSArray<RadarEvent *> * _Nullable events) {
-            if (status != RadarStatusSuccess) {
-                if (completionHandler) {
-                    [RadarUtils runOnMainThread:^{
-                        completionHandler(status, nil, nil, nil);
-                    }];
-                }
-
-                return;
-            }
++ (void)logConversionWithType:(NSString *)type
+                     metadata:(NSDictionary *_Nullable)metadata
+            completionHandler:(RadarLogConversionCompletionHandler)completionHandler {
+    NSTimeInterval lastTrackedTimeInterval = [[NSDate date] timeIntervalSinceDate:[RadarSettings lastTrackedTime]];
+    BOOL isLastTrackRecent = lastTrackedTimeInterval < 60;
 
+    CLAuthorizationStatus authorizationStatus = [[RadarLocationManager sharedInstance].permissionsHelper locationAuthorizationStatus];
+    if (authorizationStatus == kCLAuthorizationStatusDenied || isLastTrackRecent) {
+        [self sendLogConversionRequestWithType:type metadata:metadata completionHandler:completionHandler];
+        
+        return;
+    }
+    
+    [self trackOnceWithCompletionHandler:^(RadarStatus status, CLLocation * _Nullable location, NSArray<RadarEvent *> * _Nullable events, RadarUser * _Nullable user) {
+        if (status != RadarStatusSuccess) {
             if (completionHandler) {
                 [RadarUtils runOnMainThread:^{
-                    completionHandler(status, location, events, user);
+                    completionHandler(status, nil);
```

Comment:
we should still send the event correct?

Reply:
not here, since the track call fails, we don't log a conversion and instead call `completionHandler` with a non-success status and `nil` event